test/integration/acceptance/add_signatory_test.cpp

```diff
@@ -0,0 +1,172 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <gtest/gtest.h>
+#include "framework/integration_framework/integration_test_framework.hpp"
+#include "integration/acceptance/acceptance_fixture.hpp"
+
+using namespace integration_framework;
+using namespace shared_model;
+
+class AddSignatory : public AcceptanceFixture {
+ public:
+  auto makeFirstUser(const interface::RolePermissionSet &perms = {
+                         interface::permissions::Role::kAddSignatory}) {
+    return AcceptanceFixture::makeUserWithPerms(perms);
+  }
+
+  auto makeSecondUser(const interface::RolePermissionSet &perms = {
+                          interface::permissions::Role::kReceive}) {
+    return complete(
+        createUserWithPerms(kUser2, kUser2Keypair.publicKey(), kRole2, perms)
+            .creatorAccountId(kAdminId),
+        kAdminKeypair);
+  }
+
+  const std::string kRole2 = "roletwo";
+  const std::string kUser2 = "usertwo";
+  const std::string kUser2Id = kUser2 + "@test";
+  const std::string kAmount = "1.0";
+  const crypto::Keypair kUser2Keypair =
+      crypto::DefaultCryptoAlgorithmType::generateKeypair();
+};
+
+#define CHECK_BLOCK(i) \
+  [](auto &block) { ASSERT_EQ(block->transactions().size(), i); }
+
+/**
+ * C224 Add existing public key of other user
+ * @given some user with CanAddSignatory permission and a second user
+ * @when execute tx with AddSignatory where the first is a creator and the
+ *       second's key is added as a signatory
+ * @then there is the tx in proposal
+ */
+TEST_F(AddSignatory, Basic) {
```

Comment:
Maybe run a query to determine if the signatory was added?

Reply:
added